java/ql/src/Telemetry/APIUsage.qll

```diff
@@ -0,0 +1,44 @@
+import java
+private import semmle.code.java.dataflow.FlowSteps
+private import semmle.code.java.dataflow.ExternalFlow
+
+string jarName(CompilationUnit cu) {
+  result = cu.getParentContainer().toString().regexpCapture(".*/(.*\\.jar)/?.*", 1)
+}
+
+predicate isJavaRuntime(Callable call) {
+  jarName(call.getCompilationUnit()) = "rt.jar" or
+  call.getCompilationUnit().getParentContainer().toString().substring(0, 14) = "/modules/java."
+}
+
+// TODO Is this heuristic too broad?
```

Comment:
Likely yes. It certainly looks very broad.

Reply:
While very broad, this just excludes things with primitives which (as far as I can wrap my head around) can't propagate taint in any way. My initial idea was to exclude such things to reduce the number of odd calls like `String#length`, `hashcode`, `Collection,size()`, `Set.isEmpty` that are not modelled (and afaik don't need to be modelled). Any thoughts on making this more accurate?